power_grid_model_c/power_grid_model/include/power_grid_model/component/genericbranch.hpp

```diff
@@ -0,0 +1,84 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+
+#include "branch.hpp"
+
+#include "../auxiliary/input.hpp"
+#include "../auxiliary/output.hpp"
+#include "../auxiliary/update.hpp"
+#include "../calculation_parameters.hpp"
+#include "../common/common.hpp"
+#include "../common/exception.hpp"
+#include "../common/three_phase_tensor.hpp"
+
+/*
+    generic Branch: either a line N = 1 or a transformoer N = t*e^(j*theta1)
+    paramaters should be given as r1, x1, ....
+
+     -----| |-----------y1_series-------
+          | |   |                 |
+          | |   y1_shunt          y1_shunt
+          | |   |                 |
+          | |   |                 |
+     -----| |--------------------------
+          N = k * e^(j*theta1)
+
+*/
+
+namespace power_grid_model {
+
+class GenericBranch final : public Branch {
+  public:
+    using InputType = GenericBranchInput;
+    using UpdateType = BranchUpdate;
+    static constexpr char const* name = "generic_branch";
+
+    explicit GenericBranch(GenericBranchInput const& genericbranch_input, double u1_rated, double u2_rated)
+        : Branch{genericbranch_input},
+          sn_{genericbranch_input.sn},
+          r1_{genericbranch_input.r1},
+          x1_{genericbranch_input.x1},
+          g1_{genericbranch_input.g1},
+          b1_{genericbranch_input.b1},
+          k_{std::isnan(genericbranch_input.k) ? 1.0 : genericbranch_input.k},
```

Comment:
The `k_` in the class should always be off-nominal transformer ratio. Given the code you wrote here, the user input `k` is also off-nominal transformer ratio (instead of regular transformer ratio). If this is a deliberate choice, please document it.